binary/expression.proto

```diff
@@ -42,18 +42,27 @@ message Expression {
             double fp64 = 11;
             string string = 12;
             bytes binary = 13;
+            
+            // microseconds since epoch
             fixed64 timestamp = 14;
+            
+            // microseconds since epoch (at midnight)
```

Comment:
Is this really microseconds? I thought it was milliseconds. Also, why is midnight mentioned? The UNIX epoch is already defined to be midnight.

Reply:
I went with microseconds to be consistent with timestamp. Midnight is mentioned because this is a date. I would think that only certain values are valid (those at midnight)

FYI, I've abandoned these changes for now. Mostly extraneous from the more heavy lifting in this patch.